clippy_lints/src/no_effect.rs

```diff
@@ -356,8 +356,8 @@ fn reduce_expression<'a>(cx: &LateContext<'_>, expr: &'a Expr<'a>) -> Option<Vec
                 None
             }
         },
-        ExprKind::Block(block, _) => {
-            if block.stmts.is_empty() {
+        ExprKind::Block(block, label) => {
+            if block.stmts.is_empty() && label.is_none() {
```

Comment:
I wonder if this should check `!block.targeted_by_break` instead. I believe it *currently* doesn't make a difference as the lint is currently implemented, but a block without an explicit label can still be targeted by a `break`. The `?` operator in `try` blocks desugars to such a `break`.

Reply:
(I agree with y21)

Reply:
Done! I didn't know about `targeted_by_break`, thanks.